app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java

```diff
@@ -206,8 +262,38 @@ public void display(Message message) {
             implements DisplayUnreadMessagesCountInteractor.Renderer {
         @Override
         public void renderUnreadMessagesCount(int unreadMessagesCount) {
-            mPageAdapter.updateTitle(MESSAGES_CONTAINER_POSITION,
+            mPageAdapter.updateTitle(MESSAGES_CONTAINER_ID,
                     getMessagesContainerTitle(unreadMessagesCount));
         }
     }
+
+    private class KmlEntityInfoDisplayer implements DisplayKmlEntityInfoInteractor.Displayer {
+
+        @Override
+        public void display(KmlEntityInfo kmlEntityInfo) {
+            updateDetailsPage(kmlEntityInfo);
+            mView.setCurrentItem(mPageAdapter.getPosition(DETAILS_CONTAINER_ID), true);
+            updateCurrentlySelectedPageId();
+        }
+
+        @Override
+        public void hide() {
+            removeDetailsPage();
+        }
+
+        private void updateDetailsPage(KmlEntityInfo kmlEntityInfo) {
```

Comment:
displayDetailsPage
and split the method to update and add according to suited condition